### ISSUE TITLE
ArgumentOutOfRangeException when using PreserveInsertOrder

### ISSUE THREAD

**Issue by BrettRossier** [borisdj/EFCore.BulkExtensions#183]
I'm attempting to bulk insert some parent and child records, but I am getting `ArgumentOutOfRangeException: 'Specified argument was out of the range of valid values.
Parameter name: name'` when adding `PreserveInsertOrder = true` to the BulkConfig passed into BulkInsert for the parent records (inserted first).  The EventLog object and its child collection of EventLogState are coming into my outer method/class as generics with interface constraints (to allow me to access Id properties generically so I can set the child Ids after parent records are inserted), but I'm not sure how relevant that might be to this issue.

BulkInsert call:
```
DbContext.BulkInsert(logs, new BulkConfig { SetOutputIdentity = true, PreserveInsertOrder = true });
```

Stack Trace:
```
   at FastMember_dynamic.EventLog_2.get_Item(Object , String )
   at EFCore.BulkExtensions.TableInfo.UpdateEntitiesIdentity[T](IList`1 entities, IList`1 entitiesWithOutputIdentity)
   at EFCore.BulkExtensions.TableInfo.LoadOutputData[T](DbContext context, IList`1 entities)
   at EFCore.BulkExtensions.SqlBulkOperation.Merge[T](DbContext context, IList`1 entities, TableInfo tableInfo, OperationType operationType, Action`1 progress)
   at EFCore.BulkExtensions.DbContextBulkExtensions.BulkInsert[T](DbContext context, IList`1 entities, BulkConfig bulkConfig, Action`1 progress)
```

**Comment by BrettRossier** [comment]
I just dug into the code a bit, and found that the `IdentityColumnName` used by `TableInfo.UpdateEntitiesIdentity<T>` is being set to "ID".  On the database table, it is capitalized that way, but the scaffolded Entity's property is "Id".  I made the following changes in `TableInfo.UpdateEntitiesIdentity<T>` to test:
```
    accessor[entities[i], "Id"] = accessor[entitiesWithOutputIdentity[i], "Id"]; //works
    //accessor[entities[i], IdentityColumnName] = accessor[entitiesWithOutputIdentity[i], IdentityColumnName]; //doesn't work
    //accessor[entities[i], "ID"] = accessor[entitiesWithOutputIdentity[i], "ID"]; //doesn't work
```

**Comment by P-de-Jong** [comment]
@BrettRossier  This indeed seems to be the problem.. :(

**Comment by salarzobir** [comment]
Please fix this issue in next version.
as workaround i uninstalled the nuget package and download the source code then added the csproj to my solution and replaced this line:
`accessor[entities[i], IdentityColumnName] = accessor[entitiesWithOutputIdentity[i], IdentityColumnName];`
with this:
`accessor[entities[i], "Id"] = accessor[entitiesWithOutputIdentity[i], "Id"];`

Because all my pk names in entities is "Id" but in Database is "id".

**Comment by borisdj** [comment]
Fixed and NuGet 2.4.7 published
